src/test/java/io/debezium/connector/yugabytedb/consistent/MessageTest.java

```diff
@@ -34,4 +61,157 @@ void comparatorTest() {
     }
 
+    @Test
+    public void checkForBeginMessages() {
+        // Building a dummy row message
+        CdcService.RowMessage.Builder rowMessageBuilder = CdcService.RowMessage.newBuilder()
+                .setOp(CdcService.RowMessage.Op.BEGIN)
+                .setCommitTime(10L)
+                .setTransactionId(ByteString.EMPTY)
+                .setRecordTime(5L);
+        CdcService.RowMessage rowMessage = rowMessageBuilder.build();
+        CdcService.CDCSDKProtoRecordPB record = CdcService.CDCSDKProtoRecordPB.newBuilder()
+                .setRowMessage(rowMessage).build();
+
+        Message m = new Message.Builder().setRecord(record).setTabletId("dummyTabletId")
+                .setSnapshotTime(0).build();
+
+        assertTrue(Message.isBegin(m));
+    }
+
+    @Test
+    public void checkForCommitMessages() {
+        // Building a dummy row message
+        CdcService.RowMessage.Builder rowMessageBuilder = CdcService.RowMessage.newBuilder()
+                .setOp(CdcService.RowMessage.Op.COMMIT)
+                .setCommitTime(10L)
+                .setTransactionId(ByteString.EMPTY)
+                .setRecordTime(5L);
+        CdcService.RowMessage rowMessage = rowMessageBuilder.build();
+        CdcService.CDCSDKProtoRecordPB record = CdcService.CDCSDKProtoRecordPB.newBuilder()
+                .setRowMessage(rowMessage).build();
+
+        Message m = new Message.Builder().setRecord(record).setTabletId("dummyTabletId")
+                .setSnapshotTime(0).build();
+
+        assertTrue(Message.isCommit(m));
+    }
+
+    @Test
+    public void checkNeitherBeginNorCommit() {
+        // Building a dummy row message
+        CdcService.RowMessage.Builder rowMessageBuilder = CdcService.RowMessage.newBuilder()
+                .setOp(CdcService.RowMessage.Op.INSERT)
+                .setCommitTime(10L)
+                .setTransactionId(ByteString.EMPTY)
+                .setRecordTime(5L);
+        CdcService.RowMessage rowMessage = rowMessageBuilder.build();
+        CdcService.CDCSDKProtoRecordPB record = CdcService.CDCSDKProtoRecordPB.newBuilder()
+                .setRowMessage(rowMessage).build();
+
+        Message m1 = new Message.Builder().setRecord(record).setTabletId("dummyTabletId")
+                .setSnapshotTime(0).build();
+        Message m2 = new Message.Builder().setRecord(record).setTabletId("anotherDummyTablet")
+                .setSnapshotTime(0).build();
+
+        assertTrue(Message.notBeginCommit(m1, m2));
+    }
+
+    @DisplayName("Test for different combinations of compareTo")
+    @ParameterizedTest
+    @MethodSource("parameterSource")
+    public void compareToTest(CdcService.RowMessage.Op op1, CdcService.RowMessage.Op op2,
+                              long commitTime1, long commitTime2, long recordTime1,
+                              long recordTime2, long expectedResult) {
+        Params p = getParameter(op1, op2, commitTime1, commitTime2, recordTime1, recordTime2, expectedResult);
```

Comment:
Can `Params` and `getParameter` class be removed the functionality inlined? The main reason for the request is that the logic is close to the test body instead of having to scroll down looking for the class and function. It doesnt seem that big anyway to build a couple of messages and an assert statement.